lang/cs/Org.Apache.REEF.IO/FileSystem/Hadoop/HadoopFileSystem.cs

```diff
@@ -67,11 +67,8 @@ public Uri CreateUriForPath(string path)
             {
                 throw new ArgumentException("null path passed in CreateUriForPath");
             }
-
-            Uri uri;
-            uri = new Uri(path);
+            var uri = path.StartsWith(_uriPrefix) ? new Uri(path) : new Uri(_uriPrefix + '/' + path.Trim('/'));
```

Comment:
Please append with the built-in Uri class instead of path separator operations.
```csharp
return path.StartsWith(uriPrefix) ? new Uri(path) : new Uri(uriPrefix, path);
```

Reply:
Fixed.